### PR TITLE
fix(k8s): error if controller-gen fails

### PR DESCRIPTION
**Pull request by samnguyen-outreach**
## What this PR does / why we need it
- Adds the `-e` option to the bash invocation in the `go:generate` comment for Kubernetes controller code-gen
- Without the `-e`, bash will silently eat any errors from `controller-gen`

## Jira ID

[QSS-1430]

## Notes for your reviewers

Before change:
```
❯ make gogenerate
 :: Running gogenerate
github.com/getoutreach/database-provisioning-operator/api/k8s/database/v1:-: name requested for invalid type: interface{}
github.com/getoutreach/database-provisioning-operator/api/k8s/database/v1:-: invalid map value type: interface{}
Error: not all generators ran successfully
run `controller-gen object paths=./api/k8s/database/v1 -w` to see all available markers, or `controller-gen object paths=./api/k8s/database/v1 -h` for usage

❯ echo $?
0
``` 

after change:
```
❯ make gogenerate
 :: Running gogenerate
github.com/getoutreach/database-provisioning-operator/api/k8s/database/v1:-: name requested for invalid type: interface{}
github.com/getoutreach/database-provisioning-operator/api/k8s/database/v1:-: invalid map value type: interface{}
Error: not all generators ran successfully
run `controller-gen object paths=./api/k8s/database/v1 -w` to see all available markers, or `controller-gen object paths=./api/k8s/database/v1 -h` for usage
api/k8s/database/v1/v1.go:10: running "/usr/bin/env": exit status 1
make: *** [gogenerate] Error 1
```

[QSS-1430]: https://outreach-io.atlassian.net/browse/QSS-1430?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by jaredallard — APPROVED** [review]
Good catch!

**Comment by samnguyen-outreach** [comment]
> Good catch!

We were wondering why our go generate was spitting errors but not failing 🤣

File: templates/api/kubernetes/version.go.tpl [hunk 1]
```diff
@@ -10,7 +10,7 @@
 package {{ .version }}
 
 //nolint:lll //Why: Long shell script
-//go:generate /usr/bin/env bash -c "pushd ../../..{{if not (empty .package)}}/..{{end}} >/dev/null 2>&1 && ./scripts/shell-wrapper.sh gobin.sh sigs.k8s.io/controller-tools/cmd/controller-gen@v0.7.0 object paths=./api/k8s/{{ .package }}/{{ .version }}; popd >/dev/null 2>&1"
+//go:generate /usr/bin/env bash -e -c "pushd ../../..{{if not (empty .package)}}/..{{end}} >/dev/null 2>&1 && ./scripts/shell-wrapper.sh gobin.sh sigs.k8s.io/controller-tools/cmd/controller-gen@v0.7.0 object paths=./api/k8s/{{ .package }}/{{ .version }}; popd >/dev/null 2>&1"
```
Comment (jkinkead) [inline templates/api/kubernetes/version.go.tpl:13]: I think alternately:
```suggestion
//go:generate /usr/bin/env bash -c "pushd ../../..{{if not (empty .package)}}/..{{end}} >/dev/null 2>&1 && ./scripts/shell-wrapper.sh gobin.sh sigs.k8s.io/controller-tools/cmd/controller-gen@v0.7.0 object paths=./api/k8s/{{ .package }}/{{ .version }} && popd >/dev/null 2>&1"
```
(the asymmetrical `&&` is weird either way)

Comment (samnguyen-outreach) [reply templates/api/kubernetes/version.go.tpl:13]: Yup that works too.  I'll leave it up to y'all to tweak/merge as you see fit.  For now I will just manually fix in my repo.

**Pull request merged**